Automattic/jetpack
# Docs: add ToC and plugin creation guidelines to monorepo docs

#### Changes proposed in this Pull Request:
﻿- Since the document is getting long, add a Table of Contents.
- Add the steps one must follow to create a new plugin in the monorepo.

#### Jetpack product discussion

* No

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Check for typos.
* Check that the links work.
